### ISSUE TITLE
bazel testing `//...` fails on -Werror multi line comment

### ISSUE THREAD

**Issue by VivekSubr** [envoyproxy/envoy#25117]
*Title*: *bazel test fails on -Werror multi line comment*

*Description*:
Test target Compilation failure

*Repro steps*:
git clone latest envoy
bazel test //...

You'll get error message,
```
ERROR: /home/vivek/istio-1.16.1/envoy-new/envoy-built/contrib/sip_proxy/filters/network/source/BUILD:108:17: Compiling contrib/sip_proxy/filters/network/source/metadata.cc failed: (Exit 1): gcc failed: error executing command /usr/bin/gcc -U_FORTIFY_SOURCE -fstack-protector -Wall -Wunused-but-set-parameter -Wno-free-nonheap-object -fno-omit-frame-pointer '-std=c++0x' -MD -MF ... (remaining 107 arguments skipped)

Use --sandbox_debug to see verbose messages from the sandbox
contrib/sip_proxy/filters/network/source/metadata.cc:29:3: error: multi-line comment [-Werror=comment]
   29 |   // Eg: Route: <sip:test@pcsf-cfed.cncs.svc.cluster.local;role=anch;lr;transport=udp; \
      |   ^

```

**Comment by RyanTheOptimist** [comment]
I think you're saying that you get this build error when you add an extra -W to Envoy build args. Is that right?

**Comment by VivekSubr** [comment]
@RyanTheOptimist - not exactly, it's set somewhere in by the build system, all I do is bazel test //...

I have to change the comments in metadata.cc everytime.

**Comment by RyanTheOptimist** [comment]
Hm. This doesn't happen in CI and it doesn't happen for me (or other developers) locally. I wonder if your toolchain is different somehow?

cc: @phlax

**Comment by VivekSubr** [comment]
It's on WSL2 (ubuntu), Windows, I followed all steps mentioned for ubuntu on: https://github.com/envoyproxy/envoy/blob/main/DEVELOPER.md

Your CI would be a containerized build, I think, I'm not doing inside container.

**Comment by phlax** [comment]
i dont think `//...` is tested in CI and trying to run it locally it fails

The first problem is that the `mobile/` workspace is unexpected (#25114)

The next issue i hit was running with RBE - afaict the `moonjit` foreign_cc build is not hermetic/correctly set up - either way it wont build/run with RBE - so i had to abandon using it - there may well be other non-RBE issues behind that

Trying to run that command without RBE is just more of a resource commitment than i can currently make (after a couple of resource starvation failures)

So i didnt make it as far as to trigger the error you are seeing - but im wondering if we can isolate the circumstances and/or fix

One thing im noticing is that you are using GCC not LLVM to compile - that is a much less tested approach 

TLDR - i dont think you can run that command currently - but you definitely should be able to
